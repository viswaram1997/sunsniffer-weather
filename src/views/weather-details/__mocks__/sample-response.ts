export const mockResponse = {"cod":"200","message":0,"cnt":40,"list":[{"dt":1647874800,"main":{"temp":29.98,"feels_like":36.98,"temp_min":28.66,"temp_max":29.98,"pressure":1010,"sea_level":1010,"grnd_level":1006,"humidity":85,"temp_kf":1.32},"weather":[{"id":803,"main":"Clouds","description":"broken clouds","icon":"04n"}],"clouds":{"all":75},"wind":{"speed":4.95,"deg":167,"gust":7},"visibility":10000,"pop":0.06,"sys":{"pod":"n"},"dt_txt":"2022-03-21 15:00:00"},{"dt":1647885600,"main":{"temp":29.45,"feels_like":36.45,"temp_min":28.4,"temp_max":29.45,"pressure":1009,"sea_level":1009,"grnd_level":1006,"humidity":83,"temp_kf":1.05},"weather":[{"id":803,"main":"Clouds","description":"broken clouds","icon":"04n"}],"clouds":{"all":79},"wind":{"speed":4.38,"deg":152,"gust":6.03},"visibility":10000,"pop":0.06,"sys":{"pod":"n"},"dt_txt":"2022-03-21 18:00:00"},{"dt":1647896400,"main":{"temp":28.51,"feels_like":34.18,"temp_min":27.77,"temp_max":28.51,"pressure":1007,"sea_level":1007,"grnd_level":1004,"humidity":84,"temp_kf":0.74},"weather":[{"id":803,"main":"Clouds","description":"broken clouds","icon":"04n"}],"clouds":{"all":70},"wind":{"speed":4.05,"deg":180,"gust":6.59},"visibility":10000,"pop":0,"sys":{"pod":"n"},"dt_txt":"2022-03-21 21:00:00"},{"dt":1647907200,"main":{"temp":27.65,"feels_like":31.24,"temp_min":27.65,"temp_max":27.65,"pressure":1006,"sea_level":1006,"grnd_level":1004,"humidity":80,"temp_kf":0},"weather":[{"id":803,"main":"Clouds","description":"broken clouds","icon":"04n"}],"clouds":{"all":77},"wind":{"speed":4.69,"deg":190,"gust":7.85},"visibility":10000,"pop":0,"sys":{"pod":"n"},"dt_txt":"2022-03-22 00:00:00"},{"dt":1647918000,"main":{"temp":29.84,"feels_like":33.12,"temp_min":29.84,"temp_max":29.84,"pressure":1008,"sea_level":1008,"grnd_level":1007,"humidity":63,"temp_kf":0},"weather":[{"id":801,"main":"Clouds","description":"few clouds","icon":"02d"}],"clouds":{"all":19},"wind":{"speed":4.66,"deg":206,"gust":6.22},"visibility":10000,"pop":0,"sys":{"pod":"d"},"dt_txt":"2022-03-22 03:00:00"},{"dt":1647928800,"main":{"temp":32.11,"feels_like":35.87,"temp_min":32.11,"temp_max":32.11,"pressure":1008,"sea_level":1008,"grnd_level":1006,"humidity":55,"temp_kf":0},"weather":[{"id":802,"main":"Clouds","description":"scattered clouds","icon":"03d"}],"clouds":{"all":27},"wind":{"speed":5.5,"deg":147,"gust":4.68},"visibility":10000,"pop":0,"sys":{"pod":"d"},"dt_txt":"2022-03-22 06:00:00"},{"dt":1647939600,"main":{"temp":31.27,"feels_like":35.43,"temp_min":31.27,"temp_max":31.27,"pressure":1004,"sea_level":1004,"grnd_level":1003,"humidity":60,"temp_kf":0},"weather":[{"id":802,"main":"Clouds","description":"scattered clouds","icon":"03d"}],"clouds":{"all":40},"wind":{"speed":7.02,"deg":146,"gust":5.94},"visibility":10000,"pop":0,"sys":{"pod":"d"},"dt_txt":"2022-03-22 09:00:00"},{"dt":1647950400,"main":{"temp":29.28,"feels_like":33.55,"temp_min":29.28,"temp_max":29.28,"pressure":1004,"sea_level":1004,"grnd_level":1003,"humidity":71,"temp_kf":0},"weather":[{"id":803,"main":"Clouds","description":"broken clouds","icon":"04d"}],"clouds":{"all":67},"wind":{"speed":6.02,"deg":137,"gust":7.03},"visibility":10000,"pop":0,"sys":{"pod":"d"},"dt_txt":"2022-03-22 12:00:00"},{"dt":1647961200,"main":{"temp":28.27,"feels_like":32.96,"temp_min":28.27,"temp_max":28.27,"pressure":1007,"sea_level":1007,"grnd_level":1006,"humidity":81,"temp_kf":0},"weather":[{"id":804,"main":"Clouds","description":"overcast clouds","icon":"04n"}],"clouds":{"all":100},"wind":{"speed":5.39,"deg":132,"gust":7.46},"visibility":10000,"pop":0,"sys":{"pod":"n"},"dt_txt":"2022-03-22 15:00:00"},{"dt":1647972000,"main":{"temp":27.81,"feels_like":32.21,"temp_min":27.81,"temp_max":27.81,"pressure":1008,"sea_level":1008,"grnd_level":1006,"humidity":84,"temp_kf":0},"weather":[{"id":804,"main":"Clouds","description":"overcast clouds","icon":"04n"}],"clouds":{"all":100},"wind":{"speed":4.92,"deg":148,"gust":7.2},"visibility":10000,"pop":0,"sys":{"pod":"n"},"dt_txt":"2022-03-22 18:00:00"},{"dt":1647982800,"main":{"temp":26.69,"feels_like":29.7,"temp_min":26.69,"temp_max":26.69,"pressure":1006,"sea_level":1006,"grnd_level":1004,"humidity":88,"temp_kf":0},"weather":[{"id":803,"main":"Clouds","description":"broken clouds","icon":"04n"}],"clouds":{"all":73},"wind":{"speed":4.37,"deg":201,"gust":6.64},"visibility":10000,"pop":0,"sys":{"pod":"n"},"dt_txt":"2022-03-22 21:00:00"},{"dt":1647993600,"main":{"temp":26.37,"feels_like":26.37,"temp_min":26.37,"temp_max":26.37,"pressure":1006,"sea_level":1006,"grnd_level":1005,"humidity":88,"temp_kf":0},"weather":[{"id":803,"main":"Clouds","description":"broken clouds","icon":"04n"}],"clouds":{"all":83},"wind":{"speed":5.02,"deg":196,"gust":7.99},"visibility":10000,"pop":0,"sys":{"pod":"n"},"dt_txt":"2022-03-23 00:00:00"},{"dt":1648004400,"main":{"temp":28.74,"feels_like":33.63,"temp_min":28.74,"temp_max":28.74,"pressure":1009,"sea_level":1009,"grnd_level":1008,"humidity":78,"temp_kf":0},"weather":[{"id":802,"main":"Clouds","description":"scattered clouds","icon":"03d"}],"clouds":{"all":36},"wind":{"speed":5,"deg":187,"gust":6.17},"visibility":10000,"pop":0,"sys":{"pod":"d"},"dt_txt":"2022-03-23 03:00:00"},{"dt":1648015200,"main":{"temp":31.4,"feels_like":36.55,"temp_min":31.4,"temp_max":31.4,"pressure":1008,"sea_level":1008,"grnd_level":1007,"humidity":63,"temp_kf":0},"weather":[{"id":802,"main":"Clouds","description":"scattered clouds","icon":"03d"}],"clouds":{"all":45},"wind":{"speed":6.27,"deg":148,"gust":5.52},"visibility":10000,"pop":0,"sys":{"pod":"d"},"dt_txt":"2022-03-23 06:00:00"},{"dt":1648026000,"main":{"temp":30.72,"feels_like":36.03,"temp_min":30.72,"temp_max":30.72,"pressure":1005,"sea_level":1005,"grnd_level":1003,"humidity":67,"temp_kf":0},"weather":[{"id":802,"main":"Clouds","description":"scattered clouds","icon":"03d"}],"clouds":{"all":45},"wind":{"speed":7.86,"deg":142,"gust":6.53},"visibility":10000,"pop":0,"sys":{"pod":"d"},"dt_txt":"2022-03-23 09:00:00"},{"dt":1648036800,"main":{"temp":28.58,"feels_like":33.22,"temp_min":28.58,"temp_max":28.58,"pressure":1005,"sea_level":1005,"grnd_level":1004,"humidity":78,"temp_kf":0},"weather":[{"id":802,"main":"Clouds","description":"scattered clouds","icon":"03d"}],"clouds":{"all":49},"wind":{"speed":7.75,"deg":143,"gust":8.26},"visibility":10000,"pop":0,"sys":{"pod":"d"},"dt_txt":"2022-03-23 12:00:00"},{"dt":1648047600,"main":{"temp":27.55,"feels_like":31.26,"temp_min":27.55,"temp_max":27.55,"pressure":1007,"sea_level":1007,"grnd_level":1006,"humidity":82,"temp_kf":0},"weather":[{"id":804,"main":"Clouds","description":"overcast clouds","icon":"04n"}],"clouds":{"all":96},"wind":{"speed":5.94,"deg":141,"gust":8.46},"visibility":10000,"pop":0,"sys":{"pod":"n"},"dt_txt":"2022-03-23 15:00:00"},{"dt":1648058400,"main":{"temp":27.48,"feels_like":31.34,"temp_min":27.48,"temp_max":27.48,"pressure":1008,"sea_level":1008,"grnd_level":1007,"humidity":84,"temp_kf":0},"weather":[{"id":804,"main":"Clouds","description":"overcast clouds","icon":"04n"}],"clouds":{"all":95},"wind":{"speed":4.87,"deg":160,"gust":7.11},"visibility":10000,"pop":0,"sys":{"pod":"n"},"dt_txt":"2022-03-23 18:00:00"},{"dt":1648069200,"main":{"temp":27.05,"feels_like":30.25,"temp_min":27.05,"temp_max":27.05,"pressure":1006,"sea_level":1006,"grnd_level":1005,"humidity":84,"temp_kf":0},"weather":[{"id":804,"main":"Clouds","description":"overcast clouds","icon":"04n"}],"clouds":{"all":100},"wind":{"speed":4.59,"deg":181,"gust":6.66},"visibility":10000,"pop":0,"sys":{"pod":"n"},"dt_txt":"2022-03-23 21:00:00"},{"dt":1648080000,"main":{"temp":26.53,"feels_like":26.53,"temp_min":26.53,"temp_max":26.53,"pressure":1006,"sea_level":1006,"grnd_level":1005,"humidity":87,"temp_kf":0},"weather":[{"id":804,"main":"Clouds","description":"overcast clouds","icon":"04n"}],"clouds":{"all":100},"wind":{"speed":5.1,"deg":193,"gust":7.56},"visibility":10000,"pop":0,"sys":{"pod":"n"},"dt_txt":"2022-03-24 00:00:00"},{"dt":1648090800,"main":{"temp":28.66,"feels_like":32.87,"temp_min":28.66,"temp_max":28.66,"pressure":1009,"sea_level":1009,"grnd_level":1007,"humidity":75,"temp_kf":0},"weather":[{"id":804,"main":"Clouds","description":"overcast clouds","icon":"04d"}],"clouds":{"all":100},"wind":{"speed":5.81,"deg":195,"gust":7.08},"visibility":10000,"pop":0,"sys":{"pod":"d"},"dt_txt":"2022-03-24 03:00:00"},{"dt":1648101600,"main":{"temp":30.77,"feels_like":35.1,"temp_min":30.77,"temp_max":30.77,"pressure":1008,"sea_level":1008,"grnd_level":1007,"humidity":63,"temp_kf":0},"weather":[{"id":804,"main":"Clouds","description":"overcast clouds","icon":"04d"}],"clouds":{"all":86},"wind":{"speed":6.51,"deg":143,"gust":5.43},"visibility":10000,"pop":0,"sys":{"pod":"d"},"dt_txt":"2022-03-24 06:00:00"},{"dt":1648112400,"main":{"temp":30.46,"feels_like":35.15,"temp_min":30.46,"temp_max":30.46,"pressure":1005,"sea_level":1005,"grnd_level":1004,"humidity":66,"temp_kf":0},"weather":[{"id":803,"main":"Clouds","description":"broken clouds","icon":"04d"}],"clouds":{"all":52},"wind":{"speed":7.62,"deg":144,"gust":6.39},"visibility":10000,"pop":0,"sys":{"pod":"d"},"dt_txt":"2022-03-24 09:00:00"},{"dt":1648123200,"main":{"temp":28.58,"feels_like":32.33,"temp_min":28.58,"temp_max":28.58,"pressure":1005,"sea_level":1005,"grnd_level":1003,"humidity":73,"temp_kf":0},"weather":[{"id":803,"main":"Clouds","description":"broken clouds","icon":"04d"}],"clouds":{"all":64},"wind":{"speed":6.96,"deg":139,"gust":7.58},"visibility":10000,"pop":0,"sys":{"pod":"d"},"dt_txt":"2022-03-24 12:00:00"},{"dt":1648134000,"main":{"temp":27.26,"feels_like":30.77,"temp_min":27.26,"temp_max":27.26,"pressure":1007,"sea_level":1007,"grnd_level":1006,"humidity":84,"temp_kf":0},"weather":[{"id":801,"main":"Clouds","description":"few clouds","icon":"02n"}],"clouds":{"all":18},"wind":{"speed":5.46,"deg":138,"gust":7.43},"visibility":10000,"pop":0,"sys":{"pod":"n"},"dt_txt":"2022-03-24 15:00:00"},{"dt":1648144800,"main":{"temp":26.85,"feels_like":29.85,"temp_min":26.85,"temp_max":26.85,"pressure":1008,"sea_level":1008,"grnd_level":1006,"humidity":85,"temp_kf":0},"weather":[{"id":801,"main":"Clouds","description":"few clouds","icon":"02n"}],"clouds":{"all":18},"wind":{"speed":4.48,"deg":149,"gust":6.18},"visibility":10000,"pop":0,"sys":{"pod":"n"},"dt_txt":"2022-03-24 18:00:00"},{"dt":1648155600,"main":{"temp":26.12,"feels_like":26.12,"temp_min":26.12,"temp_max":26.12,"pressure":1006,"sea_level":1006,"grnd_level":1004,"humidity":87,"temp_kf":0},"weather":[{"id":801,"main":"Clouds","description":"few clouds","icon":"02n"}],"clouds":{"all":12},"wind":{"speed":3.91,"deg":193,"gust":5.27},"visibility":10000,"pop":0,"sys":{"pod":"n"},"dt_txt":"2022-03-24 21:00:00"},{"dt":1648166400,"main":{"temp":25.97,"feels_like":25.97,"temp_min":25.97,"temp_max":25.97,"pressure":1007,"sea_level":1007,"grnd_level":1005,"humidity":89,"temp_kf":0},"weather":[{"id":801,"main":"Clouds","description":"few clouds","icon":"02n"}],"clouds":{"all":18},"wind":{"speed":4.23,"deg":195,"gust":6.04},"visibility":10000,"pop":0,"sys":{"pod":"n"},"dt_txt":"2022-03-25 00:00:00"},{"dt":1648177200,"main":{"temp":28.7,"feels_like":33.15,"temp_min":28.7,"temp_max":28.7,"pressure":1009,"sea_level":1009,"grnd_level":1008,"humidity":76,"temp_kf":0},"weather":[{"id":801,"main":"Clouds","description":"few clouds","icon":"02d"}],"clouds":{"all":21},"wind":{"speed":4.76,"deg":189,"gust":5.99},"visibility":10000,"pop":0,"sys":{"pod":"d"},"dt_txt":"2022-03-25 03:00:00"},{"dt":1648188000,"main":{"temp":30.77,"feels_like":35.61,"temp_min":30.77,"temp_max":30.77,"pressure":1009,"sea_level":1009,"grnd_level":1007,"humidity":65,"temp_kf":0},"weather":[{"id":801,"main":"Clouds","description":"few clouds","icon":"02d"}],"clouds":{"all":18},"wind":{"speed":6.25,"deg":139,"gust":4.97},"visibility":10000,"pop":0,"sys":{"pod":"d"},"dt_txt":"2022-03-25 06:00:00"},{"dt":1648198800,"main":{"temp":30.24,"feels_like":35.63,"temp_min":30.24,"temp_max":30.24,"pressure":1006,"sea_level":1006,"grnd_level":1005,"humidity":70,"temp_kf":0},"weather":[{"id":801,"main":"Clouds","description":"few clouds","icon":"02d"}],"clouds":{"all":12},"wind":{"speed":7.78,"deg":140,"gust":6.71},"visibility":10000,"pop":0,"sys":{"pod":"d"},"dt_txt":"2022-03-25 09:00:00"},{"dt":1648209600,"main":{"temp":28.55,"feels_like":32.96,"temp_min":28.55,"temp_max":28.55,"pressure":1006,"sea_level":1006,"grnd_level":1005,"humidity":77,"temp_kf":0},"weather":[{"id":800,"main":"Clear","description":"clear sky","icon":"01d"}],"clouds":{"all":10},"wind":{"speed":6.78,"deg":140,"gust":7.31},"visibility":10000,"pop":0,"sys":{"pod":"d"},"dt_txt":"2022-03-25 12:00:00"},{"dt":1648220400,"main":{"temp":27.39,"feels_like":31.11,"temp_min":27.39,"temp_max":27.39,"pressure":1009,"sea_level":1009,"grnd_level":1008,"humidity":84,"temp_kf":0},"weather":[{"id":803,"main":"Clouds","description":"broken clouds","icon":"04n"}],"clouds":{"all":64},"wind":{"speed":5.69,"deg":138,"gust":7.59},"visibility":10000,"pop":0,"sys":{"pod":"n"},"dt_txt":"2022-03-25 15:00:00"},{"dt":1648231200,"main":{"temp":26.84,"feels_like":29.82,"temp_min":26.84,"temp_max":26.84,"pressure":1010,"sea_level":1010,"grnd_level":1008,"humidity":85,"temp_kf":0},"weather":[{"id":803,"main":"Clouds","description":"broken clouds","icon":"04n"}],"clouds":{"all":52},"wind":{"speed":4.44,"deg":162,"gust":6.09},"visibility":10000,"pop":0,"sys":{"pod":"n"},"dt_txt":"2022-03-25 18:00:00"},{"dt":1648242000,"main":{"temp":26.21,"feels_like":26.21,"temp_min":26.21,"temp_max":26.21,"pressure":1008,"sea_level":1008,"grnd_level":1006,"humidity":86,"temp_kf":0},"weather":[{"id":800,"main":"Clear","description":"clear sky","icon":"01n"}],"clouds":{"all":8},"wind":{"speed":4.11,"deg":194,"gust":5.76},"visibility":10000,"pop":0,"sys":{"pod":"n"},"dt_txt":"2022-03-25 21:00:00"},{"dt":1648252800,"main":{"temp":25.96,"feels_like":25.96,"temp_min":25.96,"temp_max":25.96,"pressure":1008,"sea_level":1008,"grnd_level":1007,"humidity":88,"temp_kf":0},"weather":[{"id":802,"main":"Clouds","description":"scattered clouds","icon":"03n"}],"clouds":{"all":49},"wind":{"speed":4.41,"deg":202,"gust":6.66},"visibility":10000,"pop":0,"sys":{"pod":"n"},"dt_txt":"2022-03-26 00:00:00"},{"dt":1648263600,"main":{"temp":28.53,"feels_like":32.39,"temp_min":28.53,"temp_max":28.53,"pressure":1011,"sea_level":1011,"grnd_level":1010,"humidity":74,"temp_kf":0},"weather":[{"id":803,"main":"Clouds","description":"broken clouds","icon":"04d"}],"clouds":{"all":57},"wind":{"speed":4.4,"deg":197,"gust":5.17},"visibility":10000,"pop":0,"sys":{"pod":"d"},"dt_txt":"2022-03-26 03:00:00"},{"dt":1648274400,"main":{"temp":31.03,"feels_like":34.91,"temp_min":31.03,"temp_max":31.03,"pressure":1010,"sea_level":1010,"grnd_level":1008,"humidity":60,"temp_kf":0},"weather":[{"id":802,"main":"Clouds","description":"scattered clouds","icon":"03d"}],"clouds":{"all":43},"wind":{"speed":5.55,"deg":142,"gust":4.16},"visibility":10000,"pop":0,"sys":{"pod":"d"},"dt_txt":"2022-03-26 06:00:00"},{"dt":1648285200,"main":{"temp":30.44,"feels_like":34.85,"temp_min":30.44,"temp_max":30.44,"pressure":1007,"sea_level":1007,"grnd_level":1005,"humidity":65,"temp_kf":0},"weather":[{"id":802,"main":"Clouds","description":"scattered clouds","icon":"03d"}],"clouds":{"all":48},"wind":{"speed":7.13,"deg":136,"gust":5.51},"visibility":10000,"pop":0,"sys":{"pod":"d"},"dt_txt":"2022-03-26 09:00:00"},{"dt":1648296000,"main":{"temp":28.73,"feels_like":32.49,"temp_min":28.73,"temp_max":28.73,"pressure":1007,"sea_level":1007,"grnd_level":1006,"humidity":72,"temp_kf":0},"weather":[{"id":802,"main":"Clouds","description":"scattered clouds","icon":"03d"}],"clouds":{"all":26},"wind":{"speed":6.74,"deg":138,"gust":7.13},"visibility":10000,"pop":0,"sys":{"pod":"d"},"dt_txt":"2022-03-26 12:00:00"}],"city":{"id":1264527,"name":"Chennai","coord":{"lat":13.0878,"lon":80.2785},"country":"IN","population":4328063,"timezone":19800,"sunrise":1647823358,"sunset":1647866981}}