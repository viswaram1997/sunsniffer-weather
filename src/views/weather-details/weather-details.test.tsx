import { render, act, waitFor } from "@testing-library/react";
import WeatherDetailsComponent from "./index"
const mockedUsedNavigate = jest.fn();
import axios from 'helpers';
import { mockResponse } from "./__mocks__/sample-response"
jest.mock('helpers')

const mockAxios = axios as jest.Mocked<typeof axios>;

jest.mock('react-router-dom', () => ({
    ...jest.requireActual('react-router-dom') as any,
    useNavigate: () => mockedUsedNavigate,
    useParams: () => ({ city: 'london' })
}));

describe('Weather Details page', () => {

    test('should match snapshot', async () => {

        mockAxios.get.mockResolvedValueOnce({ data: mockResponse })
        const result = render(<WeatherDetailsComponent />)
        await waitFor(() => expect(result.baseElement).toMatchSnapshot())

    })

    test('should 5 cards', async () => {

        mockAxios.get.mockResolvedValueOnce({ data: mockResponse })

        const result = render(<WeatherDetailsComponent />);
        const card = await waitFor(() => result.findAllByTestId('weather-card'))
        expect(card?.length).toEqual(5)
    })

    test('should error for city not found', async () => {
        mockAxios.get.mockRejectedValue({ response: { data: { code: 404, message: 'No city Found' } } })
        const result = render(<WeatherDetailsComponent />);
        const error = await result.findByText('No city Found')
        await waitFor(() => expect(error).not.toBeNull())
    })

});