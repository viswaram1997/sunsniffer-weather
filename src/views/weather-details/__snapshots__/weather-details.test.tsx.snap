// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Weather Details page should match snapshot 1`] = `
<body>
  <div>
    <div
      class="weather-details-wrapper"
    >
      <h2>
        Weather Forecast for Next 5days
      </h2>
      <div
        class="weather-cards-wrapper"
      >
        <div
          class="weather-header row"
        >
          <div
            class="col-md-4 col-3"
          >
            Day
          </div>
          <div
            class="col-md-2 col-3"
          >
            Humidity
          </div>
          <div
            class="col-md-3 col-3"
          >
            Weather
          </div>
          <div
            class="col-md-3 col-3"
          >
            Temprature
          </div>
        </div>
        <div
          class="loader-wrapper "
        >
          <img
            alt="loader"
            class="loader-animation"
            src="sunsniffer-loader.svg"
          />
        </div>
      </div>
    </div>
  </div>
</body>
`;
